### ISSUE TITLE
Structure

### ISSUE THREAD

**Issue by theSage21** [theSage21/ironmap#1]
There are two main models, User and Question. User can be the inbuilt Django user model. Each question has a question text (the question statement). Another model exists called TestCase. Each test case has in input string and an output string. Each test case has a relation to a question via a foreign key.

A user can login and attempt a question. What question they answer is provided by a question recommendation system. The question recommendation engine suggests a question to be solved by seeking to maximize the amount of skill set information for the user while minimizing the number of questions needed to be answered by the user. (Maybe via PCA. This is another issue).

**Comment by Shivank98** [comment]
if i got it right it is basically a webpage where user is supposed to answer a question ...?

**Comment by theSage21** [comment]
Yes. Although the answer is in code. You might want to take a look at [openJudge](https://github.com/theSage21/openJudge/). That's the kind of input we're looking for.

**Comment by Shivank98** [comment]
ohk i am looking at that.
